tests/test__collection_api.py

```diff
@@ -1313,9 +1323,90 @@ def test__create_index_wrong_type(self):
         with self.assertRaises(TypeError):
             self.db.collection.create_index([('value', 1, 'foo', 'bar')])
 
-    def test__create_ttl_index(self):
-        with self.assertRaises(NotImplementedError):
-            self.db.collection.create_index([('value', 1)], expireAfterSeconds=3600)
+    def test__ttl_index_record_expiration(self):
```

Comment:
Please split in 3 test functions (as you can see, we are happy having multiple test functions, and here you might run into trouble because one test is leaking into the other, as your `drop` call suggests)